### PR TITLE
Add a System Diagram and overview page

### PR DESCRIPTION
**Pull request by zmerlynn**
Adds a System Diagram and a whirlwind tour of the pieces of Agones.

**Special notes for your reviewer**:

Should this be in `Advanced`? I couldn't think of a better place for it.

### CODE REVIEW

File: site/static/diagrams/system-diagram.dot [hunk 1]
```diff
@@ -0,0 +1,165 @@
+digraph {
```
Comment (markmandel) [inline site/static/diagrams/system-diagram.dot:1]: Apache licence please 😄

Comment (zmerlynn) [reply site/static/diagrams/system-diagram.dot:1]: Hmm, we don't actually license any of the other .dot files?

Comment (markmandel) [reply site/static/diagrams/system-diagram.dot:1]: We probably should 😁

Comment (zmerlynn) [reply site/static/diagrams/system-diagram.dot:15]: Done, PTAL

File: site/content/en/docs/Advanced/system-diagram.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* Agones-installed Kubernetes webhooks, which handle defaulting and validation for Agones CRs,
+* and the `GameServerAllocation` `APIService`, which handles `GameServer` allocations (either from the Allocator Service or the Kubernetes API).
+
+## `agones-ping` (not pictured)
+
+`agones-ping` is a simple ping service for latency testing from your game client - see [Latency Testing]({{< relref "ping-service.md">}})).
+
+# Agones CRDs
+
+See [Create a Game Server]({{< relref "create-gameserver.md">}})), [Create a Game Server]({{< relref "create-fleet.md">}})) for examples of Agones CRDs in action, or the [API Reference]({{< ref "/docs/Reference" >}}) for more detail.
```
Comment (markmandel) [inline site/content/en/docs/Advanced/system-diagram.md:43]:
```suggestion
See [Create a Game Server]({{< relref "create-gameserver.md">}}), [Create a Game Server]({{< relref "create-fleet.md">}}) for examples of Agones CRDs in action, or the [API Reference]({{< ref "/docs/Reference" >}}) for more detail.
```

File: site/content/en/docs/Advanced/system-diagram.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## `agones-extensions`
+
+`agones-extensions` is the endpoint for:
+* Agones-installed Kubernetes webhooks, which handle defaulting and validation for Agones CRs,
+* and the `GameServerAllocation` `APIService`, which handles `GameServer` allocations (either from the Allocator Service or the Kubernetes API).
+
+## `agones-ping` (not pictured)
+
+`agones-ping` is a simple ping service for latency testing from your game client - see [Latency Testing]({{< relref "ping-service.md">}})).
```
Comment (markmandel) [inline site/content/en/docs/Advanced/system-diagram.md:39]:
```suggestion
`agones-ping` is a simple ping service for latency testing from your game client - see [Latency Testing]({{< relref "ping-service.md">}}).
```

File: site/content/en/docs/Advanced/system-diagram.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+# Agones CRDs
+
+See [Create a Game Server]({{< relref "create-gameserver.md">}})), [Create a Game Server]({{< relref "create-fleet.md">}})) for examples of Agones CRDs in action, or the [API Reference]({{< ref "/docs/Reference" >}}) for more detail.
+
+All of the Agones CRDs are controlled and updated by `agones-controller`. `GameServer` is additionally updated by the SDK Sidecar and `agones-extensions` (moving a `GameServer` from [`Ready` to `Allocated`]({{< ref "/docs/Reference/gameserver.md#gameserver-state-diagram" >}}))
```
Comment (markmandel) [inline site/content/en/docs/Advanced/system-diagram.md:45]:
```suggestion
All of the Agones CRDs are controlled and updated by `agones-controller`. `GameServer` is additionally updated by the SDK Sidecar and `agones-extensions` (moving a `GameServer` from [`Ready` to `Allocated`]({{< ref "/docs/Reference/gameserver.md#gameserver-state-diagram" >}})
```

**Review by markmandel — COMMENTED** [review]
Looks great - I just saw it had a few two many closing brackets.

As a reminder, if you need to preview locally - `make site-server` 👍🏻

**Comment by zmerlynn** [comment]
> Looks great - I just saw it had a few two many closing brackets.
> 
> As a reminder, if you need to preview locally - `make site-server` 👍🏻

Oops. PTAL!

**Comment by markmandel** [comment]
Waiting on a preview to show, but I expect it'll be good to go.

**Review by markmandel — APPROVED** [review]
Previewed it locally - looks good 👍🏻

**Pull request merged**